lnst/Recipes/ENRT/ConfigMixins/FirewallMixin.py

```diff
@@ -0,0 +1,197 @@
+from lnst.Recipes.ENRT.ConfigMixins import BaseSubConfigMixin
+from tempfile import mkstemp
+from os import close, write, unlink
+import logging
+import copy
+
+class FirewallMixin(BaseSubConfigMixin):
+    """
+    A config mixin to apply custom firewall rulesets on hosts.
+    Do not inherit directly, use one of the derived classes below instead.
+    """
+
+    @property
+    def firewall_rulesets(self):
+        """
+        This property holds a dictionary of firewall rulesets in textual
+        representation, indexed by the host it should be applied to. A typical
+        use is:
+
+        { self.matched.host1: <host1 ruleset>,
+          self.matched.host2: <host2 ruleset> }
+
+        The rulesets will be applied by a derived class's _apply_ruleset()
+        method, i.e. typically fed into 'nft -f' or 'iptables-restore'.
+        """
+        return {}
+
+    def _apply_ruleset(self, host, path):
+        raise NotImplementedError("Method _apply_ruleset must be defined by a child class.")
+
+    def _dump_ruleset(self, host, config):
+        raise NotImplementedError("Method _dump_ruleset must be defined by a child class.")
+
+    def _push_ruleset(self, host, ruleset):
+        """
+        Push `ruleset` into a temporary file on `host`, returning its path.
+        """
+        fd, name = mkstemp(prefix="FirewallMixin", text=True)
+        write(fd, ruleset.encode('utf-8'))
+        close(fd)
+        remote_path = host.copy_file_to_machine(name)
```

Comment:
We should probably remove these files in `remove_sub_configuration`. No need to fix this now, since there's no convenient method to do this through Host/Namespace API, but we should create a github issue to deal with that later.

Reply:
By introducing the FirewallControl class running on the hosts, I was able to pretty easily ellminate use of these temporary files. Many thanks to @olichtne for pointing it out, this simplified the code quite a bit!